dot/network/transaction.go

```diff
@@ -106,36 +107,59 @@ func decodeTransactionHandshake(_ []byte) (Handshake, error) {
 	return &transactionHandshake{}, nil
 }
 
-func (s *Service) createBatchMessageHandler(txnBatch chan *BatchMessage) NotificationsMessageBatchHandler {
-	return func(peer peer.ID, msg NotificationsMessage) (msgs []*BatchMessage, err error) {
+func (s *Service) createBatchMessageHandler(txnBatchCh chan *BatchMessage) NotificationsMessageBatchHandler {
+	go func() {
+		protocolID := s.host.protocolID + transactionsID
+		ticker := time.NewTicker(s.cfg.SlotDuration)
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-s.ctx.Done():
+				return
+			case <-ticker.C:
+				timeOut := time.NewTimer(s.cfg.SlotDuration / 3)
+				var completed bool
+				for !completed {
+					select {
+					case <-timeOut.C:
+						completed = true
+					case txnMsg := <-txnBatchCh:
+						propagate, err := s.handleTransactionMessage(txnMsg.peer, txnMsg.msg)
+						if err != nil {
+							s.host.closeProtocolStream(protocolID, txnMsg.peer)
+							continue
+						}
+
+						if s.noGossip || !propagate {
+							continue
+						}
+
+						if !s.gossip.hasSeen(txnMsg.msg) {
+							s.broadcastExcluding(s.notificationsProtocols[TransactionMsgType], txnMsg.peer, txnMsg.msg)
+						}
+					}
+				}
+			}
+		}
+	}()
+
+	return func(peer peer.ID, msg NotificationsMessage) {
 		data := &BatchMessage{
 			msg:  msg,
 			peer: peer,
 		}
-		txnBatch <- data
 
-		if len(txnBatch) < s.batchSize {
-			return nil, nil
-		}
+		timeOut := time.NewTimer(time.Millisecond * 200)
```

Comment:
What's the reasoning behind the 200ms value?

Maybe name it as a constant together with a comment?

Reply:
We wait `200ms` if the channel is full. Not sure what should be the exact value. Any thoughts?
Named it `txnBatchChTimeout`

Reply:
Should we have the batch size-based (event driven) on top of time-based?
In case there is a lot of messages coming in as a burst.
What do you think?

Reply:
Having event-based will cause issues because in order to validate transactions we take a lock on the state trie. Block production might suffer due to this.